### PR TITLE
Pavolum/add pii logging flag

### PR DESCRIPTION
**Pull request by pavolum**
### Purpose
*What is the context of this pull request? Why is it being done?*

Added a flag for personal data logging in the appsetting.json files for VA and Skill sample. That value is populated down to the multiple areas were it applies. Documentation has been updated as well and default value for the logging flag set to true. 

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

### CODE REVIEW

File: samples/csharp/assistants/virtual-assistant/VirtualAssistantSample/Startup.cs [hunk 1]
```diff
@@ -81,7 +84,15 @@ public void ConfigureServices(IServiceCollection services)
             services.AddSingleton<ITelemetryInitializer, OperationCorrelationTelemetryInitializer>();
             services.AddSingleton<ITelemetryInitializer, TelemetryBotIdInitializer>();
             services.AddSingleton<TelemetryInitializerMiddleware>();
-            services.AddSingleton<TelemetryLoggerMiddleware>();
+
+            if (settings.LogPersonalData)
+            {
+                services.AddSingleton<TelemetryLoggerMiddleware>(s => new TelemetryLoggerMiddleware(s.GetService<IBotTelemetryClient>(), true));
```
Comment (taicchoumsft) [inline samples/csharp/assistants/virtual-assistant/VirtualAssistantSample/Startup.cs]: >true [](start = 137, length = 4)

It looks like you can simplify this `if` statement:
```
services.AddSingleton<TelemetryLoggerMiddleware>(s => new TelemetryLoggerMiddleware(s.GetService<IBotTelemetryClient>(), settngs.LogPersonalData));
```

Comment (siddimore) [reply samples/csharp/assistants/virtual-assistant/VirtualAssistantSample/Startup.cs]: `services.AddSingleton<TelemetryLoggerMiddleware>(s => new TelemetryLoggerMiddleware(s.GetService<IBotTelemetryClient>(), Convert.ToBoolean(settngs.LogPersonalData)));`

File: samples/csharp/assistants/virtual-assistant/VirtualAssistantSample/appsettings.json [hunk 2]
```diff
@@ -21,5 +21,6 @@
   },
   "contentModerator": {
     "key": ""
-  }
+  },
+  "logPersonalInfo": true
```
Comment (siddimore) [inline samples/csharp/assistants/virtual-assistant/VirtualAssistantSample/appsettings.json:25]: default should be false

File: samples/csharp/skill/SkillSample/Startup.cs [hunk 3]
```diff
@@ -83,8 +86,14 @@ public void ConfigureServices(IServiceCollection services)
             services.AddSingleton<ITelemetryInitializer, OperationCorrelationTelemetryInitializer>();
             services.AddSingleton<ITelemetryInitializer, TelemetryBotIdInitializer>();
             services.AddSingleton<TelemetryInitializerMiddleware>();
-            services.AddSingleton<TelemetryLoggerMiddleware>();
-
+            if (settings.LogPersonalData)
+            {
+                services.AddSingleton<TelemetryLoggerMiddleware>(s => new TelemetryLoggerMiddleware(s.GetService<IBotTelemetryClient>(), true));
```
Comment (siddimore) [inline samples/csharp/skill/SkillSample/Startup.cs:91]: `services.AddSingleton(s => new TelemetryLoggerMiddleware(s.GetService(), Convert.ToBoolean(settngs.LogPersonalData)));`

File: samples/csharp/skill/SkillSample/appsettings.json [hunk 4]
```diff
@@ -17,5 +17,6 @@
     "containerId": "botstate-collection",
     "databaseId": "botstate-db"
   },
-  "properties": {}
+  "properties": {},
+  "logPersonalInfo": true
```
Comment (siddimore) [inline samples/csharp/skill/SkillSample/appsettings.json:21]: default should be false IMO

File: samples/csharp/assistants/virtual-assistant/VirtualAssistantSample/appsettings.json (hunk 2 above)
Comment (darrenj) [reply samples/csharp/assistants/virtual-assistant/VirtualAssistantSample/appsettings.json:25]: We are leaning towards on by default, we have lots of issues of people saying hey powerbi template isn't showing all this data so we think given the bot is in the creators control having utterances collected is "ok". Thoughts?

**Review by lauren-mills — APPROVED** [review]
Should this be in master or next? If master, we will need to do a release.

**Comment by pavolum** [comment]
Accidentally merging to master, closing and creating new PR merging to next

**Pull request closed without merging**